agent/bench-scripts/pbench-fio

```diff
@@ -575,7 +575,9 @@ function fio_run_job() {
 		for client in `echo $clients | sed -e s/,/" "/g`; do
 			client_opts="$client_opts --client=$client"
 			if [ ! -z $numjobs ]; then
-				let max_jobs=$numjobs
+                # so that a separate filename will be generated
```

Comment:
Can you say more as to why this works?

So we'll have multiple jobs run per client?

Reply:
max_jobs is set to 0 at the top of the loop.  So if we define numjobs in the fio command (i.e. to be something other than 1), then max_jobs will be set to numjobs, which would be wrong - we want max_jobs to be set to the total number of fio processes.  Again see fio documentation for --max-jobs.  By adding numjobs to maxjobs for each client, we arrive at the correct value for max jobs, which is the total number of fio processes in the run.  Because of "directory=$target" line, fio will auto-generate one filename for each (host, job-number) pair.

This is essentially the same logic as what happens if numjobs is *not* defined (3 lines down) - in that case, we add 1 to max_jobs for each target in the target list, but the end result is the same - max_jobs is defined to be the total number of fio processes running in the test.

Reply:
Outdated comment, see discussion in more recent code comments.